### ISSUE TITLE
Can you expose SoapParser setPagingDetails(), its a great method for paging

### ISSUE THREAD

**Issue by pingcrosby** [thuld/CrmFetchKit#1]
Hi (great library by the way)

Can you expose the SoapParser externally to enable a caller to easily use the paging cookie.  Its used internally in fetchall.

If not then could you add an example to the site showing how a user can use the paging functionality.

Attached is a trivialised (badly coded example); showing CRMFetchKit loading 2 pages of data, by utilising the SoapParser setPagingDetails() method which I made public. 

Is there a better way to update the fetchxml and call with the paging cookie and page no.??

```
 var fxml2 = [
        '<fetch count="5" returntotalrecordcount= "true" >',
        '<entity name="customeraddress" >',
        '<attribute name="line1" />',
        '<attribute name="city" />',
        '<attribute name="stateorprovince" />',
        '<attribute name="county" />',
        '<attribute name="country" />',
        '<attribute name="postalcode" />',
        '<order attribute="city" descending= "true" />',
        '</entity>',
        '</fetch>'
    ].join('');
    CrmFetchKit.FetchMore(fxml2).then(function (response) {
        console.log(response.totalRecordCount);
        console.log(response.moreRecords);
        console.log(response.entityName);
        console.log(response.pagingCookie);
        for (var i = 0; i < response.entities.length; i++) {
            console.log(response.entities[i].getValue('line1'));
        }
        var sp = CrmFetchKit.GetSoapParser();
        fxml2 = sp.setPagingDetails(fxml2, 2, response.pagingCookie);
        CrmFetchKit.FetchMore(fxml2).then(function (response) {
            console.log(response.totalRecordCount);
            console.log(response.moreRecords);
            console.log(response.entityName);
            console.log(response.pagingCookies);
            for (var i = 0; i < response.entities.length; i++) {
                console.log(response.entities[i].getValue('line1'));
            }
            var sp = CrmFetchKit.GetSoapParser();
            fxml2 = sp.setPagingDetails(fxml2, 2, response.pagingCookie);
        });
    });
```

**Comment by thuld** [comment]
Hello pingcrosby,

just to make sure I undestand you correctly: You need the paging details to fetch the data only for a specific set of pages (e.g. only records from page 1 and 2).

I could add the "fetchByPage" method tho the lib, this way I do not have to expose an internal member of the module. 

````javascript
// allows the loading of record only for certain pages (e.g. only page 1 and 2)
function fetchByPage(fetchxml, pageNumber, pagingCookie) {
    
    // add page-number & paging-cookie
    var pagingFetchXml = soapParser.setPagingDetails(
                             fetchxml, pageNumber, pagingCookie);
                             
    return fetchMore(pagingFetchXml);
    
}
````

Daniel

**Comment by pingcrosby** [comment]
Daniel

This would be a good solution, but can you also include a parameter to adjust the no. of records to return.  

Obviously i could just adjust the fetchxml before calling 'fetchMore...' but as you have already implemented the xml parser it seemed better to use your toolset.

I am using your api with jquery DataTables like this
http://datatables.net/forums/discussion/comment/75271#Comment_75271

```
       // inject the 'how..many do records do we want' page count..this is the table allowing the user to select 10,20, 30, records etc.  If this value changes then reset page cookie to null.
            fetchxml = sp.setPagingCount(fetchxml, data.length);
 
            if (pagingCookie != null) {
                var pageNumber = 1 + (data.start / data.length);
                // first time around its null after that its set and we need to pass
                fetchxml = sp.setPagingDetails(fetchxml, pageNumber, pagingCookie);
            }
            CrmFetchKit.FetchMore(fetchxml).then(function (response) {
                pagingCookie = response.pagingCookie;
}
```

**Comment by thuld** [comment]
Just released the version 3.2.0 including the "FetchByPage" method. It does not include the feature for "set-count" attriubte.
